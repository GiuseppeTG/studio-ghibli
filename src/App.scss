@import './components/Home/Home.scss';
@import './components/Navbar/Navbar.scss';
@import './components/People/People.scss';

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}
