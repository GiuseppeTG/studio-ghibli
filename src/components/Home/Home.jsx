import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { getFilms } from '../../redux/films/films';
import Filter from '../Filter/Filter';
import mainImage from '../../images/title-logo.png';

export default function Home() {
  const dispatch = useDispatch();
  const films = useSelector((state) => state.films);
  const [filtered, setFiltered] = useState([]);
  const [activeDirector, setActiveDirector] = useState('All Movies');

  useEffect(() => {
    dispatch(getFilms());
  }, []);

  return (
    <div className="homepage">
      <div className="welcome-container">
        <img src={mainImage} alt="studio ghibli logo" />
        <div className="main-text">
          <p className="app-descriprion">Here you can find all Studio Ghibli info about their films.</p>
        </div>
      </div>

      <div className="filter-container">
        <p className="filter-instruction">Filter by movie director</p>
        <Filter
          films={films}
          setFiltered={setFiltered}
          activeDirector={activeDirector}
          setActiveDirector={setActiveDirector}
        />
      </div>

      <div className="film-list-container">
        <ul className="film-list">
          {filtered.length === 0 && (

            films.map((film) => (

              <Link
                key={film.id}
                to={`/films/${film.id}`}
              >
                <div className="film-container" style={{ backgroundImage: `url(${film.image})` }} />
              </Link>
            ))
          )}
          {filtered.length !== 0 && (

            filtered.map((film) => (
              <Link
                key={film.id}
                to={`/films/${film.id}`}
              >
                <div className="film-container" style={{ backgroundImage: `url(${film.image})` }} />
              </Link>
            ))
          )}

        </ul>
      </div>
    </div>
  );
}
