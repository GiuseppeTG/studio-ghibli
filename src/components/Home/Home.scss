@import '../../scssVariables.scss';

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.homepage {
  display: flex;
  flex-direction: column;
  font-family: $poppins;
  margin-bottom: 4rem;
}

.welcome-container {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;

  img {
    width: 100%;
    max-width: 600px;
  }
}

.main-text {
  display: flex;
  flex-direction: column;
  gap: 1rem;
  text-align: center;
  font-weight: 200;
  color: rgba(255, 255, 255, 0.824);
}

.film-list-container {
  display: flex;
}

.film-list {
  display: grid;
  width: 100%;
  padding: 1rem 4rem;
  justify-items: center;
  gap: 1rem;
  grid-template-columns: repeat(auto-fit, minmax(180px, 1fr));
}

.film-container {
  width: 180px;
  height: 270px;
  background-repeat: no-repeat;
  background-size: cover;
  transition: 0.5s;
}

.film-container:hover {
  outline: white solid 1px;
  outline-offset: 0.2rem;
  transform: scale(1.3);
}

@media(max-width:768px) {
  .film-list {
    gap: 0.25rem;
    padding: 0.25rem;
    grid-template-columns: repeat(auto-fit, minmax(140px, 1fr));

  }

  .film-container {
    width: 140px;
    height: 210px;
    background-repeat: no-repeat;
    background-size: cover;
  }
}