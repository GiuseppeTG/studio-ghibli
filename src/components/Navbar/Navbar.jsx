import React, { useState } from 'react';
import { NavLink } from 'react-router-dom';
import totoroIcon from '../../images/totoroIcon.png';

export default function Navbar() {
  const [isActive, setIsActive] = useState(false);

  const handleClick = () => {
    setIsActive((isActive) => !isActive);
  };

  return (

    <nav className="navbar">
      <NavLink
        to="/"
        className="logo-text"
      >
        <div className="logo-and-text">
          <img alt="logo" src={totoroIcon} className="totoro-icon" />
          <p>
            STUDIO GHIBLI
          </p>
        </div>
      </NavLink>
      <ul className={isActive ? 'active links-list' : 'links-list'}>
        <li>
          <NavLink
            to="/"
            className="link"
            onClick={handleClick}
          >
            HOME
          </NavLink>
        </li>
        <li>
          <NavLink
            to="/people"
            className="link"
            onClick={handleClick}
          >
            PEOPLE
          </NavLink>
        </li>
      </ul>
      <button type="button" className={isActive ? 'active hamburger' : 'hamburger'} onClick={handleClick}>
        <span className="bar" />
        <span className="bar" />
        <span className="bar" />
      </button>
    </nav>
  );
}
