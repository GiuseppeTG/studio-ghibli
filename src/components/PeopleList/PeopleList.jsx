/* eslint-disable react/prop-types */
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { getPeople } from '../../redux/people/people';

export default function PeopleList() {
  const dispatch = useDispatch();
  const people = useSelector((state) => state.people);
  console.log(people);

  useEffect(() => {
    dispatch(getPeople());
  }, []);

  return (
    <>
      {
      people.map((character) => (
        <div className="character-container" key={character.id}>
          <h3 className="character-name">{character.name}</h3>
          <div className="character-info">
            <p>
              <strong>Age: </strong>
              {character.age}
            </p>
            <p>
              <strong>Gender: </strong>
              {character.gender}
            </p>
            <p>
              <strong>Eye color: </strong>
              {character.eye_color}
            </p>
            <p>
              <strong>Hair color: </strong>
              {character.hair_color}
            </p>
            <button type="button" className="go-to-movie-button">
              <Link to={character.films[0].slice(31, character.films[0].length)}>
                MOVIE
              </Link>
            </button>
          </div>
        </div>
      ))
    }
    </>
  );
}
