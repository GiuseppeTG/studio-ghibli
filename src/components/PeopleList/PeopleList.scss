@import '../../scssVariables.scss';

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.people-section {
  display: flex;
  flex-direction: column;
}

.people-title {
  margin: 2rem auto 0 auto;
  color: white;
  font-family: $poppins;
  font-weight: 300;
}

.people-list {
  width: fit-content;
  display: grid;
  justify-items: center;
  grid-template-columns: auto auto;
  list-style: none;
  margin: 2rem auto;
  box-shadow: 0 0 40px 10px rgba(0, 0, 0, 0.3);
}

.people-list li:nth-child(4n) {
  background-color: $dark-blue;
}

.people-list li:nth-child(4n+1) {
  background-color: $dark-blue;
}

.people-list li:nth-child(4n - 1) {
  background-color: $mid-blue;
}

.people-list li:nth-child(4n - 2) {
  background-color: $mid-blue;
}