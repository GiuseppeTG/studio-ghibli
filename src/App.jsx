import React from 'react';
import { Route, Routes } from 'react-router-dom';
import Navbar from './components/Navbar/Navbar';
import HomePage from './pages/homePage';
import FilmsPage from './pages/filmsPage';
import PeoplePage from './pages/peoplePage';
import './App.scss';

function App() {
  return (
    <>
      <Navbar />
      <div className="main">
        <Routes>
          <Route exact path="/" element={<HomePage />} />
          <Route exact path="/films/:id" element={<FilmsPage />} />
          <Route exact path="/people" element={<PeoplePage />} />
        </Routes>
      </div>
    </>
  );
}

export default App;
