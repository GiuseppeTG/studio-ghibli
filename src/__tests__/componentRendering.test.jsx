import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import App from '../App';
import HomePage from '../pages/homePage';
import PeoplePage from '../pages/peoplePage';
import store from '../redux/configureStore';
import Home from '../components/Home/Home';
import Navbar from '../components/Navbar/Navbar';
import PeopleList from '../components/PeopleList/PeopleList';

describe('Testing component rednering propperly', () => {
  test('renders HomePage component', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <HomePage />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
  test('renders PeoplePage component', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <PeoplePage />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
  test('renders App component', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
  test('renders Home component', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
  test('renders Navbar component', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <Navbar />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
  test('renders PeopleList component', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <PeopleList />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
