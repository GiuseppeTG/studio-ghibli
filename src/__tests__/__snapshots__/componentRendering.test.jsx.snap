// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component rednering propperly renders App component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar"
      >
        <a
          aria-current="page"
          class="logo-text active"
          href="/"
        >
          <div
            class="logo-and-text"
          >
            <img
              alt="logo"
              class="totoro-icon"
              src="totoroIcon.png"
            />
            <p>
              STUDIO GHIBLI
            </p>
          </div>
        </a>
        <ul
          class="links-list"
        >
          <li>
            <a
              aria-current="page"
              class="link active"
              href="/"
            >
              HOME
            </a>
          </li>
          <li>
            <a
              class="link"
              href="/people"
            >
              PEOPLE
            </a>
          </li>
        </ul>
        <button
          class="hamburger"
          type="button"
        >
          <span
            class="bar"
          />
          <span
            class="bar"
          />
          <span
            class="bar"
          />
        </button>
      </nav>
      <div
        class="main"
      >
        <div
          class="homepage"
        >
          <div
            class="welcome-container"
          >
            <img
              alt="studio ghibli logo"
              src="title-logo.png"
            />
            <div
              class="main-text"
            >
              <p
                class="app-descriprion"
              >
                Here you can find all Studio Ghibli info about their films.
              </p>
            </div>
          </div>
          <div
            class="filter-container"
          >
            <p
              class="filter-instruction"
            >
              Filter by director
            </p>
            <select
              class="filter-input"
              id="filteri"
            >
              <option
                value="All Movies"
              >
                All Movies
              </option>
              <option
                name="Hayao Miyazaki"
                value="Hayao Miyazaki"
              >
                Hayao Miyazaki
              </option>
              <option
                value="Isao Takahata"
              >
                Isao Takahata
              </option>
              <option
                value="Yoshifumi Kondō"
              >
                Yoshifumi Kondō
              </option>
              <option
                value="Hiroyuki Morita"
              >
                Hiroyuki Morita
              </option>
              <option
                value="Gorō Miyazaki"
              >
                Gorō Miyazaki
              </option>
              <option
                value="Hiromasa Yonebayashi"
              >
                Hiromasa Yonebayashi
              </option>
              <option
                value="Michaël Dudok de Wit"
              >
                Michaël Dudok de Wit
              </option>
            </select>
          </div>
          <p
            class="home-results"
          >
            Showing 22 results
          </p>
          <div
            class="film-list-container"
          >
            <ul
              class="film-list"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar"
    >
      <a
        aria-current="page"
        class="logo-text active"
        href="/"
      >
        <div
          class="logo-and-text"
        >
          <img
            alt="logo"
            class="totoro-icon"
            src="totoroIcon.png"
          />
          <p>
            STUDIO GHIBLI
          </p>
        </div>
      </a>
      <ul
        class="links-list"
      >
        <li>
          <a
            aria-current="page"
            class="link active"
            href="/"
          >
            HOME
          </a>
        </li>
        <li>
          <a
            class="link"
            href="/people"
          >
            PEOPLE
          </a>
        </li>
      </ul>
      <button
        class="hamburger"
        type="button"
      >
        <span
          class="bar"
        />
        <span
          class="bar"
        />
        <span
          class="bar"
        />
      </button>
    </nav>
    <div
      class="main"
    >
      <div
        class="homepage"
      >
        <div
          class="welcome-container"
        >
          <img
            alt="studio ghibli logo"
            src="title-logo.png"
          />
          <div
            class="main-text"
          >
            <p
              class="app-descriprion"
            >
              Here you can find all Studio Ghibli info about their films.
            </p>
          </div>
        </div>
        <div
          class="filter-container"
        >
          <p
            class="filter-instruction"
          >
            Filter by director
          </p>
          <select
            class="filter-input"
            id="filteri"
          >
            <option
              value="All Movies"
            >
              All Movies
            </option>
            <option
              name="Hayao Miyazaki"
              value="Hayao Miyazaki"
            >
              Hayao Miyazaki
            </option>
            <option
              value="Isao Takahata"
            >
              Isao Takahata
            </option>
            <option
              value="Yoshifumi Kondō"
            >
              Yoshifumi Kondō
            </option>
            <option
              value="Hiroyuki Morita"
            >
              Hiroyuki Morita
            </option>
            <option
              value="Gorō Miyazaki"
            >
              Gorō Miyazaki
            </option>
            <option
              value="Hiromasa Yonebayashi"
            >
              Hiromasa Yonebayashi
            </option>
            <option
              value="Michaël Dudok de Wit"
            >
              Michaël Dudok de Wit
            </option>
          </select>
        </div>
        <p
          class="home-results"
        >
          Showing 22 results
        </p>
        <div
          class="film-list-container"
        >
          <ul
            class="film-list"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing component rednering propperly renders Home component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="homepage"
      >
        <div
          class="welcome-container"
        >
          <img
            alt="studio ghibli logo"
            src="title-logo.png"
          />
          <div
            class="main-text"
          >
            <p
              class="app-descriprion"
            >
              Here you can find all Studio Ghibli info about their films.
            </p>
          </div>
        </div>
        <div
          class="filter-container"
        >
          <p
            class="filter-instruction"
          >
            Filter by director
          </p>
          <select
            class="filter-input"
            id="filteri"
          >
            <option
              value="All Movies"
            >
              All Movies
            </option>
            <option
              name="Hayao Miyazaki"
              value="Hayao Miyazaki"
            >
              Hayao Miyazaki
            </option>
            <option
              value="Isao Takahata"
            >
              Isao Takahata
            </option>
            <option
              value="Yoshifumi Kondō"
            >
              Yoshifumi Kondō
            </option>
            <option
              value="Hiroyuki Morita"
            >
              Hiroyuki Morita
            </option>
            <option
              value="Gorō Miyazaki"
            >
              Gorō Miyazaki
            </option>
            <option
              value="Hiromasa Yonebayashi"
            >
              Hiromasa Yonebayashi
            </option>
            <option
              value="Michaël Dudok de Wit"
            >
              Michaël Dudok de Wit
            </option>
          </select>
        </div>
        <p
          class="home-results"
        >
          Showing 22 results
        </p>
        <div
          class="film-list-container"
        >
          <ul
            class="film-list"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="homepage"
    >
      <div
        class="welcome-container"
      >
        <img
          alt="studio ghibli logo"
          src="title-logo.png"
        />
        <div
          class="main-text"
        >
          <p
            class="app-descriprion"
          >
            Here you can find all Studio Ghibli info about their films.
          </p>
        </div>
      </div>
      <div
        class="filter-container"
      >
        <p
          class="filter-instruction"
        >
          Filter by director
        </p>
        <select
          class="filter-input"
          id="filteri"
        >
          <option
            value="All Movies"
          >
            All Movies
          </option>
          <option
            name="Hayao Miyazaki"
            value="Hayao Miyazaki"
          >
            Hayao Miyazaki
          </option>
          <option
            value="Isao Takahata"
          >
            Isao Takahata
          </option>
          <option
            value="Yoshifumi Kondō"
          >
            Yoshifumi Kondō
          </option>
          <option
            value="Hiroyuki Morita"
          >
            Hiroyuki Morita
          </option>
          <option
            value="Gorō Miyazaki"
          >
            Gorō Miyazaki
          </option>
          <option
            value="Hiromasa Yonebayashi"
          >
            Hiromasa Yonebayashi
          </option>
          <option
            value="Michaël Dudok de Wit"
          >
            Michaël Dudok de Wit
          </option>
        </select>
      </div>
      <p
        class="home-results"
      >
        Showing 22 results
      </p>
      <div
        class="film-list-container"
      >
        <ul
          class="film-list"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing component rednering propperly renders HomePage component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="homepage"
      >
        <div
          class="welcome-container"
        >
          <img
            alt="studio ghibli logo"
            src="title-logo.png"
          />
          <div
            class="main-text"
          >
            <p
              class="app-descriprion"
            >
              Here you can find all Studio Ghibli info about their films.
            </p>
          </div>
        </div>
        <div
          class="filter-container"
        >
          <p
            class="filter-instruction"
          >
            Filter by director
          </p>
          <select
            class="filter-input"
            id="filteri"
          >
            <option
              value="All Movies"
            >
              All Movies
            </option>
            <option
              name="Hayao Miyazaki"
              value="Hayao Miyazaki"
            >
              Hayao Miyazaki
            </option>
            <option
              value="Isao Takahata"
            >
              Isao Takahata
            </option>
            <option
              value="Yoshifumi Kondō"
            >
              Yoshifumi Kondō
            </option>
            <option
              value="Hiroyuki Morita"
            >
              Hiroyuki Morita
            </option>
            <option
              value="Gorō Miyazaki"
            >
              Gorō Miyazaki
            </option>
            <option
              value="Hiromasa Yonebayashi"
            >
              Hiromasa Yonebayashi
            </option>
            <option
              value="Michaël Dudok de Wit"
            >
              Michaël Dudok de Wit
            </option>
          </select>
        </div>
        <p
          class="home-results"
        >
          Showing 22 results
        </p>
        <div
          class="film-list-container"
        >
          <ul
            class="film-list"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="homepage"
    >
      <div
        class="welcome-container"
      >
        <img
          alt="studio ghibli logo"
          src="title-logo.png"
        />
        <div
          class="main-text"
        >
          <p
            class="app-descriprion"
          >
            Here you can find all Studio Ghibli info about their films.
          </p>
        </div>
      </div>
      <div
        class="filter-container"
      >
        <p
          class="filter-instruction"
        >
          Filter by director
        </p>
        <select
          class="filter-input"
          id="filteri"
        >
          <option
            value="All Movies"
          >
            All Movies
          </option>
          <option
            name="Hayao Miyazaki"
            value="Hayao Miyazaki"
          >
            Hayao Miyazaki
          </option>
          <option
            value="Isao Takahata"
          >
            Isao Takahata
          </option>
          <option
            value="Yoshifumi Kondō"
          >
            Yoshifumi Kondō
          </option>
          <option
            value="Hiroyuki Morita"
          >
            Hiroyuki Morita
          </option>
          <option
            value="Gorō Miyazaki"
          >
            Gorō Miyazaki
          </option>
          <option
            value="Hiromasa Yonebayashi"
          >
            Hiromasa Yonebayashi
          </option>
          <option
            value="Michaël Dudok de Wit"
          >
            Michaël Dudok de Wit
          </option>
        </select>
      </div>
      <p
        class="home-results"
      >
        Showing 22 results
      </p>
      <div
        class="film-list-container"
      >
        <ul
          class="film-list"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing component rednering propperly renders Navbar component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar"
      >
        <a
          aria-current="page"
          class="logo-text active"
          href="/"
        >
          <div
            class="logo-and-text"
          >
            <img
              alt="logo"
              class="totoro-icon"
              src="totoroIcon.png"
            />
            <p>
              STUDIO GHIBLI
            </p>
          </div>
        </a>
        <ul
          class="links-list"
        >
          <li>
            <a
              aria-current="page"
              class="link active"
              href="/"
            >
              HOME
            </a>
          </li>
          <li>
            <a
              class="link"
              href="/people"
            >
              PEOPLE
            </a>
          </li>
        </ul>
        <button
          class="hamburger"
          type="button"
        >
          <span
            class="bar"
          />
          <span
            class="bar"
          />
          <span
            class="bar"
          />
        </button>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar"
    >
      <a
        aria-current="page"
        class="logo-text active"
        href="/"
      >
        <div
          class="logo-and-text"
        >
          <img
            alt="logo"
            class="totoro-icon"
            src="totoroIcon.png"
          />
          <p>
            STUDIO GHIBLI
          </p>
        </div>
      </a>
      <ul
        class="links-list"
      >
        <li>
          <a
            aria-current="page"
            class="link active"
            href="/"
          >
            HOME
          </a>
        </li>
        <li>
          <a
            class="link"
            href="/people"
          >
            PEOPLE
          </a>
        </li>
      </ul>
      <button
        class="hamburger"
        type="button"
      >
        <span
          class="bar"
        />
        <span
          class="bar"
        />
        <span
          class="bar"
        />
      </button>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing component rednering propperly renders PeopleList component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="people-section"
      >
        <h2
          class="people-title"
        >
          Human Characters
        </h2>
        <ul
          class="people-list"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="people-section"
    >
      <h2
        class="people-title"
      >
        Human Characters
      </h2>
      <ul
        class="people-list"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing component rednering propperly renders PeoplePage component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="people-section"
      >
        <h2
          class="people-title"
        >
          Human Characters
        </h2>
        <ul
          class="people-list"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="people-section"
    >
      <h2
        class="people-title"
      >
        Human Characters
      </h2>
      <ul
        class="people-list"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
